### PR TITLE
[RFR] Split application file into different files

### PR DESCRIPTION
**Pull request by midays**
each file is dedicated to an IDE specific class

https://issues.redhat.com/browse/WINDUP-3639

### CODE REVIEW

File: src/lib/IDE/Intellij.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    def close_ide(self):
+        """
+        Closes the IDE
+        """
+        self.click_element(locator_type="image", locator="file_menu.png")
+        self.press_keys("up")
+        self.press_keys("enter")
+        time.sleep(1)
+        self.press_keys("enter")
+        # try:
```
Comment (nachandr) [inline src/lib/IDE/Intellij.py]: @midays , could you review these lines ? It looks like an oversight to me.

Comment (midays) [reply src/lib/IDE/Intellij.py]: When exiting the IDE there was issues with the exit button, for some reason the framework fails to locate it ,so I replaced it with "enter" click, and i added 1 second delay to wait for the popup to completely load

Comment (sshveta) [reply src/lib/IDE/Intellij.py]: @midays How is it exiting the IDE . I do not see any click on exit button .

File: src/lib/IDE/Intellij.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        #     self.wait_find_element(locator_type="image", locator="confirm_exit.png", timeout=5.0)
+        #     self.click_element(locator_type="image", locator="exit_button.png")
```
Comment (sshveta) [inline src/lib/IDE/Intellij.py]: Can we remove these comments?

File: src/lib/IDE/Intellij.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        # Provide mta cli path
+        try:
+            # self.click_element(locator_type="image", locator="mta_cli_input.png")
+            self.click(add_project_buttons[0])
+            time.sleep(2)
+            for letter in config_data["mta_cli_path"]:
+                self.type_text(letter)
+            self.press_keys("enter")
+        except Exception as exc:
+            logging.debug("MTA cli path is already present ! {}".format(exc))
```
Comment (sshveta) [inline src/lib/IDE/Intellij.py:120]: This method is very big.
We can create fixtures or utility methods for smaller tasks like :  


fill_cli_path()
fill_target()
wait_for_analysis_to_complete()  etc

**Review by sshveta — CHANGES_REQUESTED** [review]

File: src/lib/IDE/Intellij.py (hunk 1 above)
Comment (midays) [reply src/lib/IDE/Intellij.py]: it clicks on the file button in the menu > move the arrow up ( which will move the highlight to the last button i.e. exit button in the menu ), then click enter, then wait for 1 sec, then click enter again which clicks the exit button in the confirmation popup

File: src/lib/IDE/Intellij.py (hunk 3 above)
Comment (midays) [reply src/lib/IDE/Intellij.py:120]: noted, will create future PR's for this

Comment (midays) [reply src/lib/IDE/Intellij.py:120]: jira ticket created: https://issues.redhat.com/browse/WINDUP-3662

File: src/lib/IDE/Intellij.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        self.press_keys("enter")
+        time.sleep(3)
+        self.wait_find_element(locator_type="image", locator="mta_config_page_opened.png")
+        # Region defining the configuration name
+        config_name_region = self.define_region(778, 248, 1125, 284)
+        add_project_locator = self.image_locator("add_project_button.png")
+        add_project_buttons = self.find_elements(add_project_locator)
+        # Provide mta cli path
+        try:
+            # self.click_element(locator_type="image", locator="mta_cli_input.png")
```
Comment (nachandr) [inline src/lib/IDE/Intellij.py]: This comment needs to be removed.

File: src/lib/IDE/VisualStudioCode.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        # Wait for 'New Configuration' to become visible
+        time.sleep(5)
+        # Create new analysis configuration by clicking 'New Configuration' button
+        self.wait_find_element(locator_type="image", locator="create_new_config.png")
+        self.click_element(locator_type="image", locator="create_new_config.png")
+        # Wait for configuration page to load
+        time.sleep(3)
+
+        # Add project info to configuration
+        add_project_locator = self.image_locator("add_button.png")
```
Comment (nachandr) [inline src/lib/IDE/VisualStudioCode.py:97]: The image_locator() method was removed from the base class , but was not added in the VSCode child class.

File: src/lib/IDE/Intellij.py (hunk 4 above)
Comment (midays) [reply src/lib/IDE/Intellij.py]: done

File: src/lib/IDE/VisualStudioCode.py (hunk 5 above)
Comment (midays) [reply src/lib/IDE/VisualStudioCode.py:97]: done

**Review by sshveta — APPROVED** [review]

**Pull request merged**